aruco: build warnings (MSVS 2015)

relates #3240

Logs: http://pullrequest.opencv.org/buildbot/builders/precommit_windows64/builds/100075
Logs (2): http://pullrequest.opencv.org/buildbot/builders/master-contrib-win64-vc14/builds/100000

> C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\memory(901): warning C4244: 'argument': conversion from 'const int' to 'float', possible loss of data (compiling source file C:\build\precommit_windows64\4.x\opencv_contrib\modules\aruco\src\charuco.cpp) [C:\build\precommit_windows64\build\modules\aruco\opencv_aruco.vcxproj]
C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\memory(901): warning C4244: 'argument': conversion from 'const double' to 'float', possible loss of data (compiling source file C:\build\precommit_windows64\4.x\opencv_contrib\modules\aruco\test\test_boarddetection.cpp) [C:\build\precommit_windows64\build\modules\aruco\opencv_test_aruco.vcxproj]
C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\memory(901): warning C4244: 'argument': conversion from 'const int' to 'float', possible loss of data (compiling source file C:\build\precommit_windows64\4.x\opencv_contrib\modules\aruco\test\test_boarddetection.cpp) [C:\build\precommit_windows64\build\modules\aruco\opencv_test_aruco.vcxproj]

Check full compiler log for details.

Bug, found by GCC 12.1 (Fedora 36):

```
[3032/3540] Building CXX object modules/aruco/CMakeFiles/opencv_aruco.dir/src/aruco_detector.cpp.o
/home/alalek/projects/opencv/contrib/modules/aruco/src/aruco_detector.cpp: In member function ‘bool cv::aruco::DetectorParameters::writeDetectorParameters(const cv::Ptr<cv::FileStorage>&)’:
/home/alalek/projects/opencv/contrib/modules/aruco/src/aruco_detector.cpp:66:36: warning: ‘this’ pointer is null [-Wnonnull]
   66 |     if (fs.empty() && !fs->isOpened())
      |                        ~~~~~~~~~~~~^~
```

--- a/modules/aruco/src/aruco_detector.cpp
+++ b/modules/aruco/src/aruco_detector.cpp
@@ -63,7 +63,7 @@ bool DetectorParameters::readDetectorParameters(const FileNode& fn) {
 
 bool DetectorParameters::writeDetectorParameters(const Ptr<FileStorage>& fs)
 {
-    if (fs.empty() && !fs->isOpened())
+    if (fs.empty() || !fs->isOpened())
         return false;
     return readWrite(*this, nullptr, fs);
 }

--- a/modules/aruco/src/charuco.cpp
+++ b/modules/aruco/src/charuco.cpp
@@ -428,7 +428,7 @@ void detectCharucoDiamond(InputArray _image, InputArrayOfArrays _markerCorners,
         // try to find the rest of markers in the diamond
         vector< int > acceptedIdxs;
         Ptr<Board> _b = _charucoDiamondLayout.staticCast<Board>();
-        Ptr<RefineParameters> refineParameters = makePtr<RefineParameters>(minRepDistance, -1, false);
+        Ptr<RefineParameters> refineParameters = makePtr<RefineParameters>(minRepDistance, -1.f, false);
         ArucoDetector detector(dictionary, DetectorParameters::create(), refineParameters);
         detector.refineDetectedMarkers(grey, _b, currentMarker, currentMarkerId, candidates, noArray(), noArray(),
                                        acceptedIdxs);

--- a/modules/aruco/test/test_boarddetection.cpp
+++ b/modules/aruco/test/test_boarddetection.cpp
@@ -170,7 +170,7 @@ class CV_ArucoRefine : public cvtest::BaseTest {
         params->cornerRefinementMethod = aruco::CORNER_REFINE_SUBPIX;
         if (arucoAlgParams == ArucoAlgParams::USE_ARUCO3)
             params->useAruco3Detection = true;
-        Ptr<aruco::RefineParameters> refineParams = makePtr<aruco::RefineParameters>(10, 3., true);
+        Ptr<aruco::RefineParameters> refineParams = makePtr<aruco::RefineParameters>(10.f, 3.f, true);
         detector = aruco::ArucoDetector(dictionary, params, refineParams);
     }
 

